### PR TITLE
user signup

### PR DESCRIPTION
**Pull request by pabloferro**
## Summary
User signup

## Trello Card
https://trello.com/c/ghlOetot/2-crear-registro-de-usuarios

### CODE REVIEW

File: src/app/app.routes.js [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
         views: {
           main: {
-            templateUrl: '../app/layouts/state1/main.html'
+            templateUrl: '../app/layouts/home/main.html'
           }
         }
       })
-      .state('state1.index', {
-        url: '/state1',
+      .state('home.index', {
```
Comment (sbalay) [inline src/app/app.routes.js]: Do you think this state needs to be abstract? Ask me for help if needed

File: src/app/app.routes.js [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
         views: {
           main: {
-            templateUrl: '../app/layouts/state2/main.html'
+            templateUrl: '../app/layouts/signup/main.html'
           }
         }
       })
-      .state('state2.index', {
-        url: '/state2',
+      .state('signup.index', {
```
Comment (sbalay) [inline src/app/app.routes.js]: Do you think this state needs to have an abstract state as parent? Ask me for help if needed

File: src/app/app.routes.js (hunk 1 above)
Comment (pabloferro) [inline src/app/app.routes.js]: No, I can keep only the home and signup states.

File: src/app/components/home/home.jade [hunk 3]
```diff
@@ -0,0 +1,4 @@
+div
+  h1
+    | WBooks
+  a(ui-sref='signup.index') Signup
```
Comment (sbalay) [inline src/app/components/home/home.jade]: As a convention, we split the element from its content in different lines. Is easier to read.
```
a(ui-sref='signup.index')
  | Signup
```

File: src/app/components/signup/signup.jade [hunk 4]
```diff
@@ -0,0 +1,24 @@
+div(ng-controller='SignupController as signup')
```
Comment (sbalay) [inline src/app/components/signup/signup.jade]: Ui-router provides another way of assigning controllers to states, and avoid this extra wrapper. 
https://github.com/angular-ui/ui-router/wiki#controllers

File: src/app/components/signup/signup.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.frontend-bootstrap-component-1 {
```
Comment (sbalay) [inline src/app/components/signup/signup.scss]: remove this file if its unneccesary

File: src/app/components/signup/signupController.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+angular.module('pf-angular').controller('SignupController', ['$state', 'userService', function ($state, userService) {
```
Comment (sbalay) [inline src/app/components/signup/signupController.js]: As convention, we split the module's declaration in three lines.
```
angular.module('pf-angular').controller('SignupController', [ // module name
  '$state', 'userService', // module dependencies
  function ($state, userService) { // module declaration
```

File: src/app/components/signup/signupController.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  this.user = {};
```
Comment (sbalay) [inline src/app/components/signup/signupController.js]: You don't need this initialization

**Comment by sbalay** [comment]
Looks great

**Comment by sbalay** [comment]
Check those jade files. Change them to follow our convention of splitting the content from the element in different lines

**Comment by sbalay** [comment]
Looks great! :+1:

**Pull request merged**